lib/cache.ml

```diff
@@ -0,0 +1,88 @@
+open Bos
+open Rresult
+open Astring
+
+let valid_remote_name = String.map (function '/' | '@' | ':' -> '-' | x -> x)
+
+let commit_branch_name ~commit = "commit-" ^ commit
+
+let cache_branch_name ~ref = "branch-" ^ ref
+
+let home () =
+  try Sys.getenv "HOME"
+  with Not_found -> (
+    try (Unix.getpwuid (Unix.getuid ())).Unix.pw_dir
+    with Unix.Unix_error _ | Not_found ->
+      if Sys.win32 then try Sys.getenv "AppData" with Not_found -> "" else "" )
```

Comment:
The `if Sys,win32` should be done before trying `Unix.getpwuid`.

Reply:
I'll fix that!